pkg/common/config/aws_config.go

```diff
@@ -25,7 +49,49 @@ var (
 	AWSVPCSubnetIDs = "config.aws.vpcSubnetIDs"
 )
 
-func InitAWSViper() {
+func InitAWSViper() error {
+	_ = viper.BindEnv(AWSSharedCredentials, "AWS_SHARED_CREDENTIALS")
+	RegisterSecret(AWSSharedCredentials, "aws-shared-credentials")
+	sharedCreds := viper.GetString(AWSSharedCredentials)
+
+	viper.SetDefault(AWSCredentialsFile, os.Getenv("HOME")+"/.aws/osde2e/credentials")
+	_ = viper.BindEnv(AWSCredentialsFile, "AWS_CREDENTIAL_FILE")
+	customCredsPath := viper.GetString(AWSCredentialsFile)
+
+	if sharedCreds != "" {
+		// If shared credntials file is provided in env vars, it should contain two profiles named "profile-a" and profile-b"
+		// Osde2e will use one of them based on current week.
+		// While one profile is in use, the other is cleaned up using AWS nuke
+		if err := os.MkdirAll(filepath.Dir(customCredsPath), os.FileMode(0o755)); err != nil {
+			return fmt.Errorf("Could not write given shared credentials file: %v", err)
```

Comment:
Last thing and I'm good to go:

```suggestion
			return fmt.Errorf("Could not write given shared credentials file: %w", err)
```

nit, be sure to use the `%w` directive when you can to wrap errors

Reply:
(There are multiple instances of this to fix)